### PR TITLE
all: move to macaroon.v2-unstable

### PR DESCRIPTION
**Pull request by rogpeppe**
There's still a dependency on macaroon.v1 because one
of the charmrepo params uses it. That will change when
we can move juju to using macaroon.v2.

We delete the agent tests because they were a hassle to
update and the agent package is going to be deleted in the next PR.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1917/
Test PASSed.

**Review by jrwren — APPROVED** [review]
This is great. It reads like a how-to-better-use-bakery.

File: internal/v5/api.go [hunk 1]
```diff
@@ -114,10 +125,13 @@ func New(pool *charmstore.Pool, config charmstore.ServerParams, rootPath string)
 	var permChecker permChecker
 	if config.IdentityAPIURL != "" && config.AgentUsername != "" && config.AgentKey != nil {
 		logger.Infof("group membership enabled at %s; agent %s", config.IdentityAPIURL, config.AgentUsername)
-		identityClient := idmclient.New(idmclient.NewParams{
+		identityClient, err := idmclient.New(idmclient.NewParams{
 			BaseURL: config.IdentityAPIURL,
 			Client:  agent.NewClient(config.AgentUsername, config.AgentKey),
 		})
+		if err != nil {
+			panic(err)
```
Comment (mhilton) [inline internal/v5/api.go:133]: Strictly speaking I'd say that as the IdentityAPIURL originally came from user input, the error case here cannot be attributed to being a bug and so a panic isn't appropriate. I suspect we should rethink the idmclient API if we need this to complete without errors.

On the other hand I know how hard it is to fail to parse a URL, so I'm not that worried about it.

**Review by mhilton — APPROVED** [review]
LGTM with one thought

File: internal/v5/api.go (hunk 1 above)
Comment (rogpeppe) [reply internal/v5/api.go:133]: Yes. I think I'll do it in the next PR if that's OK.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**